frontend/src/lib/k8s/apiProxy.ts

```diff
@@ -800,14 +804,18 @@ export async function apply(body: KubeObjectInterface): Promise<JSON> {
   }
 
   const apiEndpoint = resourceDefToApiFactory(bodyToApply);
+  const resourceVersion = bodyToApply.metadata.resourceVersion;
 
   try {
+    delete bodyToApply.metadata.resourceVersion;
     return await apiEndpoint.post(bodyToApply);
   } catch (err) {
     // Check to see if failed because the record already exists.
     // If the failure isn't a 409 (i.e. Confilct), just rethrow.
     if ((err as ApiError).status !== 409) throw err;
 
+    // Preserve the resourceVersion if its an update request
+    bodyToApply.metadata.resourceVersion = resourceVersion;
```

Comment:
We should avoid modifying the original structure. One way to avoid it, just like we do on line 797 is to clone the body (maybe we can clone it for all cases even...).

Reply:
updated by cloning at source